testing/cli-e2etest/testscenarios/analyzer/list_analyzer_test.go

```diff
@@ -92,15 +92,16 @@ func TestListAnalyzer(t *testing.T) {
 		result := tracetestcli.Exec(t, "list analyzer --sortBy name --output json", tracetestcli.WithCLIConfig(cliConfig))
 		helpers.RequireExitCodeEqual(t, result, 0)
 
-		analyzerYAML := helpers.UnmarshalJSON[[]types.AnalyzerResource](t, result.StdOut)
-
-		require.Len(analyzerYAML, 1)
-		require.Equal("Analyzer", analyzerYAML[0].Type)
-		require.Equal("current", analyzerYAML[0].Spec.Id)
-		require.Equal("analyzer", analyzerYAML[0].Spec.Name)
-		require.True(analyzerYAML[0].Spec.Enabled)
-		require.Equal(analyzerYAML[0].Spec.MinimumScore, 95)
-		require.Len(analyzerYAML[0].Spec.Plugins, 3)
+		analyzerList := helpers.UnmarshalJSON[types.ResourceList[types.AnalyzerResource]](t, result.StdOut)
```

Comment:
this scenario was incorrect (as well as the old implementation). it was returning an array of resources instead of a `ResourceList`

Reply:
@schoren I think that's expected, the CLI is expected to return an array of resources:
```
---
resource1
---
resource2
---
resourc3
```

As per the requirements, you can see some of that from the CLI improvements epic
![Screenshot 2023-06-28 at 8 10 30](https://github.com/kubeshop/tracetest/assets/11051031/58765558-ba25-4515-8690-596a88bd5cea)

Reply:
@xoscar The originally defined specs (environments for example https://github.com/kubeshop/tracetest/issues/2182#issuecomment-1490727214) defined the json resposne to be a resource list. This allows direct passthrough from the server

Reply:
So then, how should it work? `ResourceList` for `json` and plane array list for `yaml`?

Reply:
@kdhamric had some comments about this first time we implement it as a `ResourceList`

Reply:
The use case for each formmat are different. 

The YAML list is almost surely used to export a list into a file to later import on another instace, hence the multi-document format. 

JSON is mainly machine readable, so its most likely be used in scripting. In this case, you might need extra info. 

for this same reason JSON uses the augmented verison and YAML doesn’t.

Reply:
IIRC @kdhamric comment was around the yaml format and how that should be multidocument instead of a ResourceList. This is still the case here. 

This change is only for the JSON format.

Reply:
@xoscar @kdhamric I've added a loom about the output formats so we're all on the same page on what this PR proposes

Reply:
Perfect, that makes sense, thanks for the video!